document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/PostgresQueryParser.java

```diff
@@ -1,104 +1,114 @@
 package org.hypertrace.core.documentstore.postgres.query.v1;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
-import org.hypertrace.core.documentstore.expression.type.FilterTypeExpression;
+import lombok.Getter;
 import org.hypertrace.core.documentstore.postgres.Params;
 import org.hypertrace.core.documentstore.postgres.Params.Builder;
 import org.hypertrace.core.documentstore.postgres.query.v1.vistors.PostgresFilterTypeExpressionVisitor;
 import org.hypertrace.core.documentstore.postgres.query.v1.vistors.PostgresGroupTypeExpressionVisitor;
 import org.hypertrace.core.documentstore.postgres.query.v1.vistors.PostgresSelectTypeExpressionVisitor;
 import org.hypertrace.core.documentstore.query.Pagination;
 import org.hypertrace.core.documentstore.query.Query;
-import org.hypertrace.core.documentstore.query.SelectionSpec;
 import org.hypertrace.core.documentstore.query.SortingSpec;
 
 public class PostgresQueryParser {
   private static String NOT_YET_SUPPORTED = "Not yet supported %s";
-  private Builder paramsBuilder;
   private final String collection;
 
-  public PostgresQueryParser(String collection) {
-    this.collection = collection;
+  @Getter private Builder paramsBuilder;
+
+  @Getter private Query query;
+
+  // map of alias name to parsed expression
+  // e.g qty : COUNT(DISTINCT CAST(document->>'quantity' AS NUMERIC))
+  @Getter private Map<String, String> pgSelections;
+
+  private void resetParser(Query query) {
```

Comment:
Nit: 
This looks like a setter method for `query` with some custom logic.
Can this method be named `setQuery` instead?

Reply:
No need due to - https://github.com/hypertrace/document-store/pull/95#discussion_r917748266